File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/AuthIpSecurity.java
package uk.gov.justice.digital.hmpps.oauth2server.security;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.security.web.util.matcher.IpAddressMatcher;
import org.springframework.stereotype.Component;
import uk.gov.justice.digital.hmpps.oauth2server.utils.IpAddressHelper;

import javax.servlet.http.HttpServletRequest;
import java.util.Set;

@Slf4j
@Component
public class AuthIpSecurity {

    private final Set<String> whitelist;

    public AuthIpSecurity(@Value("${application.authentication.ui.whitelist}") final Set<String> whitelist) {
        this.whitelist = whitelist;
    }

    public boolean check(final HttpServletRequest request) {
        final var remoteIp = IpAddressHelper.retrieveIpFromRemoteAddr(request);
        final var matchIp = whitelist.stream().anyMatch(ip -> new IpAddressMatcher(ip).matches(remoteIp));
        if (!matchIp) {
            log.warn("Client IP {}, is not in whitelist {}", remoteIp, whitelist);
        }
        return matchIp;
    }

}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/AuthIpSecurity.kt
package uk.gov.justice.digital.hmpps.oauth2server.security

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.security.web.util.matcher.IpAddressMatcher
import org.springframework.stereotype.Component
import uk.gov.justice.digital.hmpps.oauth2server.utils.IpAddressHelper.retrieveIpFromRemoteAddr
import javax.servlet.http.HttpServletRequest

@Component
class AuthIpSecurity(@Value("\${application.authentication.ui.allowlist}") private val allowlist: Set<String>) {
  companion object {
    private val log = LoggerFactory.getLogger(this::class.java)
  }

  fun check(request: HttpServletRequest?): Boolean {
    val remoteIp = retrieveIpFromRemoteAddr(request!!)
    val matchIp = allowlist.any { ip: String? -> IpAddressMatcher(ip).matches(remoteIp) }
    if (!matchIp) {
      log.warn("Client IP {}, is not in allowlist {}", remoteIp, allowlist)
    }
    return matchIp
  }
}
